test/integration/ResourceLockCleanup.test.ts

```diff
@@ -0,0 +1,84 @@
+import fetch from 'cross-fetch';
+import type { App, DataAccessorBasedStore, FileSystemResourceLocker, Initializable } from '../../src';
+import { readableToString, BasicRepresentation } from '../../src';
+import { describeIf, getPort } from '../util/Util';
+import { getDefaultVariables, getTestConfigPath, getTestFolder, instantiateFromConfig, removeFolder } from './Config';
+
+const port = getPort('ResourceLockCleanup');
+const baseUrl = `http://localhost:${port}/`;
+const rootFilePath = getTestFolder(`resource-lock-cleanup`);
+const resourceIdentifier = { path: `${baseUrl}container1/test.txt` };
+
+const configs: [string, any][] = [
+  [
+    'file-based', {
+      config: 'server-file.json',
+      init: async(initializable: Initializable): Promise<void> => initializable.initialize(),
+      teardown: async(): Promise<void> => removeFolder(rootFilePath),
+    }],
+  [
+    'redis-based', {
+      config: 'server-redis-lock.json',
+      init: jest.fn(),
+      teardown: jest.fn(),
+    }],
+];
+
+describeIf('docker').each(configs)('A server using %s locking',
+  (id, { config, init, teardown }): void => {
+    let app: App;
+    let store: DataAccessorBasedStore;
+    let locker: FileSystemResourceLocker;
+
+    beforeAll(async(): Promise<void> => {
+      const variables = {
+        ...getDefaultVariables(port, baseUrl),
+        'urn:solid-server:default:variable:rootFilePath': rootFilePath,
+      };
+
+      // Create the server
+      const instances = await instantiateFromConfig(
+        'urn:solid-server:test:Instances',
+        [
+          getTestConfigPath(config),
+        ],
+        variables,
+      ) as Record<string, any>;
+      ({ app, store, locker } = instances);
+      // Create the test resource
+      await store.setRepresentation(resourceIdentifier, new BasicRepresentation('abc', 'text/plain'));
+
+      // Perform additional initialization, if configured
+      await init(locker);
```

Comment:
I still don't fully understand why this is needed though https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1342#discussion_r907116713

What component exactly is complaining about this lack of root folder?

Reply:
The extra initialization here is needed when testing the `FileSystemResourceLocker`. As part of the test, we need to be able to create a lock before starting the server. If we do not call `locker.initialize()` and as a consequence create the lock folder, the subsequent `locker.acquire(..)` call cannot be completed.

Reply:
Ah right, makes sense now. I thought it was still to make the `setRepresentation` succeed but the order has changed now.